implementation/query-runner/src/main/java/net/opentsdb/tsd/QueryConfig.java

```diff
@@ -343,13 +359,169 @@ void stop(final int status_code) {
     }
   }
 
+  class CurlRunner implements Runnable {
+    final int index;
+    
+    CurlRunner(final int index) {
+      this.index = index;
+    }
+    
+    @Override
+    public void run() {
+      try {
+        final String temp_file = runner.curl_temp + id + "_" + index + ".log";
+        File f = new File(temp_file);
+        if (f.exists()) {
+          try {
+            f.delete();
+          } catch (Exception e) {
+            LOG.error("Failed to remove temp file: " + f);
+          }
+        }
+        LOG.info("Sending CURL query " + id  + " to endpoint " 
+            + shuffled_endpoints.get(index));
+        final ProcessBuilder builder = new ProcessBuilder();
+        final List<String> args = Lists.newArrayList();
+        args.add(runner.curl_exec);
+        args.add("-sS"); // suppress the progress bar but keep errors.
+        //args.add("-vv"); // for debugging
+        args.add("-w");
+        args.add("\"" + TsdbQueryRunner.CURL_STATUS + runner.curl_metrics + "\"");
+        args.add("-o");
+        args.add(temp_file);
+        args.add("-X");
+        args.add("POST");
+        args.add("-H");
+        args.add("Content-Type: application/json");
+        args.add("-d");
+        args.add("@" + runner.curl_temp + id + ".json");
+        args.add("-m");
+        args.add(Integer.toString(runner.timeout / 1000));
+        if (!Strings.isNullOrEmpty(runner.curl_flags)) {
+          // TODO - handle quotes if we need em.
+          final String[] parts = runner.curl_flags.split(" ");
+          for (int i = 0; i < parts.length; i++) {
+            args.add(parts[i]);
+          }
+        }
+        args.add(shuffled_endpoints.get(index));
+        builder.command(args);
+        builder.redirectErrorStream(true);
```

Comment:
Nice, this would merge the error stream into inputstream so we don't have to consume that separately